sds/src/secondary_validation/romanian_personal_numeric_code.rs

```diff
@@ -0,0 +1,79 @@
+use crate::secondary_validation::Validator;
+pub struct RomanianPersonalNumericCode;
+
+const CHECKSUM_WEIGHTS: &[u32] = &[2, 7, 9, 1, 4, 6, 3, 5, 8, 2, 7, 9];
+
+impl Validator for RomanianPersonalNumericCode {
+    fn is_valid_match(&self, regex_match: &str) -> bool {
+        if !regex_match.is_ascii() {
+            return false;
+        }
+        for c in regex_match.chars() {
+            if !c.is_ascii_digit() {
+                return false;
+            }
+        }
+
+        let mut calculated_checksum = 0;
+        for (i, c) in regex_match.chars().take(12).enumerate() {
```

Comment:
We could use `CHECKSUM_WEIGHTS.length()` instead of `12` here to avoid any issue between the two constants